providers/implementations/encode_decode/encode_key2text.c

```diff
@@ -348,17 +339,13 @@ static int dsa_to_text(BIO *out, const void *key, int selection)
 
     return 1;
 }
+
+# define dsa_input_type         "DSA"
 #endif
 
 /* ---------------------------------------------------------------------- */
 
 #ifndef OPENSSL_NO_EC
```

Comment:
Do you need to include the other parameters anywhere? It looks like it can gone from being OSSL_KEYMGMT_SELECT_ALL_PARAMETERS to anything that is not public or private ??

Reply:
Well, keys are conceptually composed of, at most, a public key, a private key, and parameters.  Do you imagine some other components sometime in the future?

Reply:
I am referring to OSSL_KEYMGMT_SELECT_OTHER_PARAMETERS

Reply:
It is used in rsa_to_text()

Reply:
It would be much easier if you commented on the lines you're talking about :wink:
I'll have a look

Reply:
Ah, now I see...  well, this runs with the assumption that most callers would use `OSSL_KEYMGMT_SELECT_ALL_PARAMETERS`, and I believe I've changed all our uses in libcrypto to do exactly that.  However, if someone uses OSSL_ENCODE directly, an uses, say, OSSL_KEYMGMT_SELECT_KEYPAIR | OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS, what should happen?

Considering that, at least here, the selection bits are more of a wishlist ("please give me what you can of these"), an encoder implementation can only be so strict.  Where do we draw the line?

Reply:
Its ok as it is - I just wanted you to think about the answer..